src/TwigComponent/doc/index.rst

```diff
@@ -1058,6 +1058,143 @@ Exclude specific attributes:
       My Component!
     </div>
 
+.. __Customize your component:
+
+CVA (Class variant authority)
+-----------------------------
+
+You can use the cva function to define variants for your component.
+
+.. code-block:: html+twig
+
+    {# templates/components/Alert.html.twig #}
+    {% props color = 'blue', size = 'md', class = '' %}
+
+     {% set alert = cva('alert', {
+        'color': {
+            'blue': 'bg-blue',
+            'red': 'bg-red',
+            'green': 'bg-green',
+        },
+        'size': {
+            'sm': 'text-sm',
+            'md': 'text-md',
+            'lg': 'text-lg',
+        }
+    }) %}
+
+    <div {{ attributes.defaults(
+        class: alert.apply({color, size}, class)
```

Comment:
The base class is simply a set of classes present in all variations of your components: for example, for a button, you may want all your variants to contain the class 'btn'. It's the first argument of the cva function. I added some doc for it! Thanks!

Reply:
cva twig functions first parameter are the `base` classes, they are always applied and get combined with the variants - I think having that option is great. It could be done as in your example as well, but what if you also want to ask for additional classes from "outside" (via the `<twig:X class="...">`? 
It could be concatenated with the 2nd parameter of apply() for sure, but the second param of apply is ment as "additional classes"

Base classes are part of CVA and even tho it would be solvable without `base` classes, shouldn't we stick to how CVA works in JS? https://cva.style/docs/api-reference

`defaultVariants` were already left out, because those can be handled by default values in props that are passed as variants, but do we really also want to scratch `base` classes?